### PR TITLE
Fix update bug

### PR DESCRIPTION
**Pull request by ivorcmx**
Fixed updateTaskCommand bug to not allow invalid date.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/UpdateTaskCommandParserTest.java [hunk 1]
```diff
@@ -52,17 +55,22 @@ public void parse_invalidPreamble_failure() {
...
-        assertParseFailure(parser, "1" + " t/", MESSAGE_INVALID_FORMAT); // empty deadline
+        assertParseFailure(parser, "1" + " d/", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+            UpdateTaskCommand.MESSAGE_EMPTY_PARAMETERS)); // empty description
+        assertParseFailure(parser, "1" + " t/", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+            UpdateTaskCommand.MESSAGE_EMPTY_PARAMETERS)); // empty deadline
 
         // valid description followed by invalid deadline
-        assertParseFailure(parser, "1" + DESC_SWIMMING + INVALID_DEADLINE_YEAR, MESSAGE_INVALID_FORMAT);
+        assertParseFailure(parser, "1" + DESC_SWIMMING + INVALID_DEADLINE_YEAR,
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, UpdateTaskCommand.MESSAGE_USAGE));
```
Comment (putaojuice) [inline src/test/java/seedu/address/logic/parser/UpdateTaskCommandParserTest.java:65]: In your `parse` method you used 
`throw new ParseException("Invalid date input!\n" + UpdateTaskCommand.MESSAGE_USAGE);` where the error message is the "Invalid date ... "
and in the test you use `MESSAGE_INVALID_COMMAND_FORMAT`, does the test pass in this case?

**Review by putaojuice — COMMENTED** [review]
Please try running the tests.

File: src/test/java/seedu/address/logic/parser/UpdateTaskCommandParserTest.java (hunk 1 above)
Comment (ivorcmx) [reply src/test/java/seedu/address/logic/parser/UpdateTaskCommandParserTest.java:65]: Hi @putaojuice. In this parse failure, the message thrown will beMESSAGE_INVALID_COMMAND_FORMAT as checkDeadlineFormat will be the one throwing the error. Tests passed 👍

Comment (putaojuice) [reply src/test/java/seedu/address/logic/parser/UpdateTaskCommandParserTest.java:65]: Alright, thanks for clarifying!

**Review by putaojuice — APPROVED** [review]

**Pull request merged**